test/blackbox-tests/test-cases/pkg/multiple-opam-repos.t

```diff
@@ -31,10 +31,10 @@ We have to define both repositories in the workspace, but will only use `new`.
   > (lang dune 3.10)
   > (repository
   >  (name new)
-  >  (source "git+file://$(pwd)/mock-opam-repository"))
+  >  (source "$(pwd)/mock-opam-repository"))
   > (repository
   >  (name old)
-  >  (source "git+file://$(pwd)/old-mock-opam-repository"))
+  >  (source "$(pwd)/old-mock-opam-repository"))
```

Comment:
How do we know that this is a git repo now?

Reply:
We only support git repos here.

Reply:
Shouldn't we support repositories defined on the file system as well?

Reply:
You can do that via the CLI, do you think it would be sensible to do that in `dune-workspace` as well?

Reply:
> Edit: The only way to use paths is via the CLI.

I see. I guess that's TODO. Since we need to support paths here, I wonder if we should just leave that `git+https` prefix. Otherwise, I don't see how we'll differentiate the two.

Reply:
Yes, that's a good point. Another option would be to check if it is a path and then check if the path is a git repo and if not use it as regular path, but it is a bit magical and I am not sure if that wouldn't lead to a confusing UX.

Reply:
Yes, that would definitely be confusing. Especially given that one can point to a local git repo and expect one of the two behaviors:

1. Ignore untracked/unstaged changes. Corresponds to treating it like a git repo
2. Take those changes into account. Corresponds to treating it like a normal directory

So this should be explicit

Reply:
Ok, makes sense; I'll add the distinction back. I'm sure I can coerce OpamUrl to give me more useful outputs or just parse it myself :)

Reply:
Ok, I believe this should work. If OPAM suggests HTTP as backend, we pass it to git anyway, so `https://github.com/ocaml/opam-repository.git` will be treated the same as `git+https://github.com/ocaml/opam-repository.git`.